Microsoft.HealthVault/Thing/ThingQuery.cs

```diff
@@ -1075,35 +1075,35 @@ private void AddFilterSection(XmlWriter writer)
                     writer.WriteEndElement();
                 }
 
-                if (_createdDateMin != null)
+                if (_createdDateMin != null && _createdDateMin.HasValue)
                 {
```

Comment:
nit: Each of these could read a little better and be (very slightly) more performant by changing them to "nullableDate?.HasValue == true" or "nullableDate?.Value != null"